CorsixTH/Lua/dialogs/staff_dialog.lua

```diff
@@ -109,7 +109,7 @@ function UIStaff:UIStaff(ui, staff)
       self:addColourPanel(32, 141, 171, 39, 85, 202, 219)  -- Hides Skills
     end
     self:addPanel(307, 106, 226):makeButton(0, 0, 50, 50, 308, self.fireStaff):setTooltip(_S.tooltip.staff_window.sack)
-    self:addPanel(309, 164, 226):makeButton(0, 0, 37, 50, 310, self.placeStaff):setTooltip(_S.tooltip.staff_window.pick_up)
+    self:addPanel(309, 164, 226):makeButton(0, 0, 37, 50, 310, self.pickupStaff):setTooltip(_S.tooltip.staff_window.pick_up)
```

Comment:
As the button is being amended outside of draw, it may be safe to include an afterload to close the window.

Reply:
> As the button is being amended outside of draw, it may be safe to include an afterload to close the window.

this is related to backward compatibility? 
for case to load the game with already drawn window?

Reply:
Like I know that :p

The "persistable:foo" comment is used by the save system to identify the function that should get the saved data. Also, instantiated classes are saved, so the method functions work too.

In both cases of course, you get the persisted data. New variables of the function won't exist thus.

Reply:
> Yeah the button in memory has a call embedded now pointing to a now nonexistent function.

@lewri I have made experiment:

1. Before that change: Run game. Open doctor menu. Made safe.
2. Closed game. Made this change.
3. Run game. Load save.
4. Pressed grab button on menu which was on screen after game load - nothing happened (button don't work).
5. Gamelog empty. Lua not crashed - game run smoothly.
6. Close/open doctor menu. Then button work ok.

Is that results mean that change is safe and no any backward compatibility additions required?

Reply:
You're likely still loading the old code. It's pretty trivial just to tell the game to close the window to be safe.

Reply:
Actually ignore that comment. The change you made now no longer runs the risk afaik.

Reply:
> You're likely still loading the old code.

If still loading the old code when why grab spamming before change is reproducible and after change grab spamming is not reproducible?

Reply:
`Warning: No handler for button click` occurs in the console w/debug if you change function name from what it expects, with the window open at save time. (you can load this save in your test environment to check made in 0.67 [pickup.zip](https://github.com/user-attachments/files/17355465/pickup.zip))

As a result of minimising the error please either:
1. Edit the function `UIStaff:placeStaff()` without changing function name. This doesn't need an afterload. ***or***
2. Add an afterload:
```lua
function UIStaff:afterLoad(old, new)
  if old < xxx then
    self:close()
  end
  Window.afterLoad(self, old, new)
end
```

Reply:
> 2. Add an afterload:
> 
> ```lua
>   if old < xxx then
> ```

Thanks, now I got it. 👍
What should be posted instead of 'xxx'?

Reply:
`195` for now, bumping app.lua too to 195.

Depending on #2639 it may go to `205`.